src/Control/Email/Email.php

```diff
@@ -3,766 +3,618 @@
-     * Create a new email.
-     *
-     * @param string|null $from
-     * @param string|null $to
+     * Email constructor.
+     * @param string|array|null $from
+     * @param string|array|null $to
      * @param string|null $subject
      * @param string|null $body
-     * @param string|null $bounceHandlerURL
-     * @param string|null $cc
-     * @param string|null $bcc
+     * @param string|array|null $cc
+     * @param string|array|null $bcc
+     * @param string|null $returnPath
      */
     public function __construct(
         $from = null,
         $to = null,
         $subject = null,
         $body = null,
-        $bounceHandlerURL = null,
         $cc = null,
-        $bcc = null
+        $bcc = null,
+        $returnPath = null
     ) {
-
-        if ($from !== null) {
-            $this->from = $from;
-        }
-        if ($to !== null) {
-            $this->to = $to;
-        }
-        if ($subject !== null) {
-            $this->subject = $subject;
-        }
-        if ($body !== null) {
-            $this->body = $body;
-        }
-        if ($cc !== null) {
-            $this->cc = $cc;
-        }
-        if ($bcc !== null) {
-            $this->bcc = $bcc;
-        }
-
-        if ($bounceHandlerURL !== null) {
-            Deprecation::notice('4.0', 'Use "emailbouncehandler" module');
-        }
+        $this->setFrom($from)
+            ->setTo($to)
+            ->setSubject($subject)
+            ->setBody($body)
+            ->setCC($cc)
+            ->setBCC($bcc)
+            ->setReturnPath($returnPath);
 
         parent::__construct();
     }
 
     /**
-     * Get the mailer.
-     *
-     * @return Mailer
+     * @return Swift_Message
      */
-    public static function mailer()
+    public function getSwiftMessage()
     {
-        return Injector::inst()->get('SilverStripe\\Control\\Email\\Mailer');
+        if (!$this->swiftMessage) {
+            $this->setSwiftMessage(new Swift_Message());
+        }
+
+        return $this->swiftMessage;
     }
 
     /**
-     * Attach a file based on provided raw data.
+     * @param Swift_Message $swiftMessage
      *
-     * @param string $data The raw file data (not encoded).
-     * @param string $attachedFilename Name of the file that should appear once it's sent as a separate attachment.
-     * @param string|null $mimeType MIME type to use when attaching file. If not provided, will attempt to infer via HTTP::get_mime_type().
      * @return $this
      */
-    public function attachFileFromString($data, $attachedFilename, $mimeType = null)
+    public function setSwiftMessage($swiftMessage)
     {
-        $this->attachments[] = array(
-            'contents' => $data,
-            'filename' => $attachedFilename,
-            'mimetype' => $mimeType,
-        );
+        $swiftMessage->setDate(DBDatetime::now()->Format('U'));
+        if ($defaultFrom = $this->config()->admin_email) {
+            $swiftMessage->setFrom($defaultFrom);
+        }
```

Comment:
Won’t this mean that if you use this method, you’re forced to use `admin_email` (if set)? i.e. if I set a default from address, how can I override it on a per-email basis? I think this needs to check if a from address has already been set